stacks-signer/src/client/stackerdb.rs

```diff
@@ -103,14 +104,21 @@ impl StackerDB {
             } else {
                 warn!("Chunk rejected by stackerdb: {:?}", chunk_ack);
             }
-            if let Some(reason) = chunk_ack.reason {
-                // TODO: fix this jankiness. Update stackerdb to use an error code mapping instead of just a string
-                // See: https://github.com/stacks-network/stacks-blockchain/issues/3917
-                if reason.contains("Data for this slot and version already exist") {
-                    warn!("Failed to send message to stackerdb due to wrong version number {}. Incrementing and retrying...", slot_version);
-                } else {
-                    warn!("Failed to send message to stackerdb: {}", reason);
-                    return Err(ClientError::PutChunkRejected(reason));
+            if let Some(code) = chunk_ack.code {
```

Comment:
Just out of curiousity, did it not work to do 
 `let code: StackerDBErrorCodes = serde_json::from_str(&chunk_ack.reason)?`
 
 I think I tried this myself (thought it would work because I think reason is serialized from a StackerDBErrorCode to begin with) and it didn't work but just want to confirm with you too.

Reply:
That doesn't work because `StackerDBErrorCodes` doesn't implement `Deserialize` - this `reason` comes from the custom method `code.into_json()`